File: apps/mobile/src/services/sentry.ts
/**
 * Sentry Configuration for AmPac Mobile
 * 
 * To enable Sentry:
 * 1. Install: npx expo install @sentry/react-native
 * 2. Create a Sentry project at https://sentry.io
 * 3. Add EXPO_PUBLIC_SENTRY_DSN to your .env file
 * 4. For source maps, add SENTRY_AUTH_TOKEN to your EAS secrets
 */

// Sentry is optional - check if it's installed
let Sentry: any = null;
try {
    Sentry = require('@sentry/react-native');
} catch (e) {
    console.log('📊 @sentry/react-native not installed, error reporting disabled');
}

const SENTRY_DSN = process.env.EXPO_PUBLIC_SENTRY_DSN;

export const initSentry = () => {
    if (!Sentry || !SENTRY_DSN) {
        console.log('📊 Sentry not configured, error reporting disabled');
        return;
    }

    Sentry.init({
        dsn: SENTRY_DSN,
        
        // Set environment
        environment: __DEV__ ? 'development' : 'production',
        
        // Enable performance monitoring
        tracesSampleRate: __DEV__ ? 1.0 : 0.2,
        
        // Only send errors in production
        enabled: !__DEV__,
        
        // Attach user info
        beforeSend(event: any) {
            // You can filter or modify events here
            return event;
        },
    });

    console.log('📊 Sentry initialized');
};

/**
 * Capture an exception with optional context
 */
export const captureException = (
    error: Error,
    context?: Record<string, any>
) => {
    if (!Sentry || !SENTRY_DSN) {
        console.error('Error:', error, context);
        return;
    }

    Sentry.withScope((scope: any) => {
        if (context) {
            scope.setExtras(context);
        }
        Sentry.captureException(error);
    });
};

/**
 * Capture a message for logging
 */
export const captureMessage = (
    message: string,
    level: 'info' | 'warning' | 'error' = 'info'
) => {
    if (!Sentry || !SENTRY_DSN) {
        console.log(`[${level}]`, message);
        return;
    }

    Sentry.captureMessage(message, level);
};

/**
 * Set user context for error tracking
 */
export const setUser = (user: { id: string; email?: string; username?: string } | null) => {
    if (!Sentry || !SENTRY_DSN) return;
    Sentry.setUser(user);
};

/**
 * Add breadcrumb for debugging
 */
export const addBreadcrumb = (
    message: string,
    category: string,
    data?: Record<string, any>
) => {
    if (!Sentry || !SENTRY_DSN) return;
    
    Sentry.addBreadcrumb({
        message,
        category,
        data,
        level: 'info',
    });
};

export { Sentry };

File: apps/brain/app/core/sentry.py
"""
Sentry Configuration for AmPac Brain API

To enable Sentry:
1. Create a Sentry project at https://sentry.io
2. Add SENTRY_DSN to your .env file
3. Errors will be automatically captured and reported
"""

import sentry_sdk
from sentry_sdk.integrations.fastapi import FastApiIntegration
from sentry_sdk.integrations.starlette import StarletteIntegration
from sentry_sdk.integrations.logging import LoggingIntegration
import logging
from app.core.config import get_settings

logger = logging.getLogger(__name__)

def init_sentry():
    """Initialize Sentry error tracking."""
    settings = get_settings()
    sentry_dsn = getattr(settings, 'SENTRY_DSN', None)
    
    if not sentry_dsn:
        logger.info("📊 Sentry DSN not configured, error reporting disabled")
        return
    
    sentry_sdk.init(
        dsn=sentry_dsn,
        
        # Set environment
        environment=settings.ENV,
        
        # Enable performance monitoring
        traces_sample_rate=0.1 if settings.ENV == "production" else 1.0,
        
        # Capture 100% of errors
        sample_rate=1.0,
        
        # Integrations
        integrations=[
            FastApiIntegration(transaction_style="endpoint"),
            StarletteIntegration(transaction_style="endpoint"),
            LoggingIntegration(
                level=logging.INFO,
                event_level=logging.ERROR
            ),
        ],
        
        # Don't send PII by default
        send_default_pii=False,
        
        # Release version
        release=f"ampac-brain@{settings.VERSION}",
    )
    
    logger.info("📊 Sentry initialized for error tracking")


def capture_exception(error: Exception, context: dict = None):
    """Capture an exception with optional context."""
    settings = get_settings()
    sentry_dsn = getattr(settings, 'SENTRY_DSN', None)
    
    if not sentry_dsn:
        logger.error(f"Error: {error}", exc_info=True)
        return
    
    with sentry_sdk.push_scope() as scope:
        if context:
            for key, value in context.items():
                scope.set_extra(key, value)
        sentry_sdk.capture_exception(error)


def capture_message(message: str, level: str = "info"):
    """Capture a message for logging."""
    settings = get_settings()
    sentry_dsn = getattr(settings, 'SENTRY_DSN', None)
    
    if not sentry_dsn:
        logger.log(getattr(logging, level.upper(), logging.INFO), message)
        return
    
    sentry_sdk.capture_message(message, level=level)


def set_user(user_id: str, email: str = None):
    """Set user context for error tracking."""
    sentry_sdk.set_user({
        "id": user_id,
        "email": email,
    })
